docker-compose.yml

```diff
@@ -122,6 +122,45 @@ services:
     labels:
       logs: "promtail"
 
+### Web Client
+  hyperswitch-web:
+    ports:
+      - "9050:9050"
+      - "9060:9060"
+      - "5252:5252"
+    build:
+      context: ./docker
+      dockerfile: web.Dockerfile
+    env_file:
+      - ./.env
```

Comment:
@prasad89 can you add this env file if it is needed, I'd prefer all variables in the below `environment` block tho?

Reply:
We've set up a configuration block with default variable pointers. 
```yaml
 - HYPERSWITCH_SERVER_URL=${HYPERSWITCH_SERVER_URL:-http://hyperswitch-server:8080}
 - HYPERSWITCH_CLIENT_URL=${HYPERSWITCH_CLIENT_URL:-http://localhost:9050}
 - SELF_SERVER_URL=${SELF_SERVER_URL:-http://localhost:5252}
```
For confidentiality, `HYPERSWITCH_PUBLISHABLE_KEY` and `HYPERSWITCH_SECRET_KEY` are considered personal secrets and should be stored in a .env file. 

Therefore, the complete .env file should appear as follows:

```yaml
HYPERSWITCH_PUBLISHABLE_KEY=HYPERSWITCH_PUBLISHABLE_KEY
HYPERSWITCH_SECRET_KEY=HYPERSWITCH_SECRET_KEY
HYPERSWITCH_SERVER_URL=http://hyperswitch-server:8080
HYPERSWITCH_CLIENT_URL=http://localhost:9050
SELF_SERVER_URL=http://localhost:5252
```

Reply:
but then you are specifying these keys here as well,
```
    environment:
      - HYPERSWITCH_PUBLISHABLE_KEY=$HYPERSWITCH_PUBLISHABLE_KEY
      - HYPERSWITCH_SECRET_KEY=$HYPERSWITCH_SECRET_KEY
prasad89 marked this conversation as resolved.
      - HYPERSWITCH_SERVER_URL=${HYPERSWITCH_SERVER_URL:-http://hyperswitch-server:8080}
      - HYPERSWITCH_CLIENT_URL=${HYPERSWITCH_CLIENT_URL:-http://localhost:9050}
      - SELF_SERVER_URL=${SELF_SERVER_URL:-http://localhost:5252}
```

in that case we can remove the .env file

Reply:
In our configuration, we're defining the `environment` block and specifying that `HYPERSWITCH_PUBLISHABLE_KEY` and `HYPERSWITCH_SECRET_KEY` are required for running Docker Compose, indicated by the presence of the `'$` symbol. For other variables, Docker Compose will use values from the `.env` file if provided, otherwise, it will proceed with default value from `environment` block

We are using `interpolation` method here.
REF : https://docs.docker.com/compose/environment-variables/env-file/#interpolation

Reply:
```
environment:
  - HYPERSWITCH_PUBLISHABLE_KEY=$HYPERSWITCH_PUBLISHABLE_KEY
  - HYPERSWITCH_SECRET_KEY=$HYPERSWITCH_SECRET_KEY
```

oh I was confused by these part then? where would this be used?

Reply:
Yes, those should be present in the `.env` file in order to run docker compose.

Reply:
@lsampras hope I explained you well.
Let me know if there are anymore changes.

Reply:
yeah, 
can you make the env file optional as mentioned [here](https://docs.docker.com/compose/environment-variables/set-environment-variables/#additional-information-2).
Also it seemed that control-center was facing some problems when loading the sdk for test payment..
<img width="1422" alt="image" src="https://github.com/juspay/hyperswitch/assets/21202349/3b79cf8a-fb29-4ee0-aef5-857db12afe44">

Can you test that flow once on your end and lmk if you are able to make payments through that?
you can configure a test connector for the same.

Reply:
@lsampras have tested the flow added one fix for the same. 
![Screenshot from 2024-04-26 20-04-43](https://github.com/juspay/hyperswitch/assets/67261499/943b54e2-9875-4e39-9039-74b1ea2ade37)

Reply:
@prasad89 Can you make sure the SDK loads in control-center
sharing a reference screenshot from our sandbox environment (app.hyperswitch.io) 


<img width="1728" alt="image" src="https://github.com/juspay/hyperswitch/assets/21202349/e6ac6b85-8318-48c6-a740-c8741782e9db">